docs/docs/module-api/executorch-bindings.md

```diff
@@ -0,0 +1,105 @@
+---
+title: ExecuTorch Bindings
+sidebar_position: 1
+---
+
+ExecuTorch bindings provide streamlined interface to access the [Module API](https://pytorch.org/executorch/stable/extension-module.html) directly from Javascript.
+
+:::caution[Caution]
+These bindings are primarily intended for custom model integration where no dedicated hook exists. If you are considering using a provided model, first verify whether a dedicated hook is available. Dedicated hooks simplify the implementation process by managing necessary pre and post-processing automatically. Utilizing these can save you effort and reduce complexity, ensuring you do not implement additional handling that is already covered.
+:::
+
+## Intializing ExecuTorch Module
+
+You can initialize the ExecuTorch module in your JavaScript application using the `useExecutorchModule` hook. This hook facilitates the loading of models from the specified source and prepares them for use.
+
+```typescript
+import { useExecutorchModule } from 'react-native-executorch';
+
+const executorchModule = useExecutorchModule({
+  modelSource: require('../assets/models/model.pte'),
+});
+```
+
+The `modelSource` parameter expects a location string pointing to the model binary. For more details on how to specify model sources, refer to the [loading models](../fundamentals/loading-models.md) documentation.
+
+### Arguments
+
+**`modelSource`** - A string that specifies the location of the model binary.
+
+### Returns
+
+| Field               | Type                                                       | Description                                                                                                                                       |
+| ------------------- | ---------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `error`             | <code>ETError &#124; null</code>                           | Contains the error message if the model failed to load.                                                                                           |
+| `isModelGenerating` | `boolean`                                                  | Indicates whether the model is currently generating.                                                                                              |
+| `isModelReady`      | `boolean`                                                  | Indicates whether the model is ready.                                                                                                             |
```

Comment:
In the API it is called isModelLoading, as we agreed on isModelReady, we have to change it, just a note